<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Sean Davis" />


<title>Querying TCGA data using BigQuery</title>


<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="site_libs/highlight/default.css"
      type="text/css" />
<script src="site_libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>


<link rel="stylesheet" href="/Library/Frameworks/R.framework/Versions/3.3/Resources/library/BiocStyle/resources/html/bioconductor.css" type="text/css" />

<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if(links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script>

</head>

<body>


<div id="header">
<h1 class="title">Querying TCGA data using BigQuery</h1>
<h4 class="author"><em>Sean Davis</em></h4>
<h4 class="date"><em>May 5, 2016</em></h4>
</div>

<h1>Contents</h1>
<div id="TOC">
<ul>
<li><a href="#background"><span class="toc-section-number">1</span> Background</a><ul>
<li><a href="#what-is-this-bigquery"><span class="toc-section-number">1.1</span> What is this BigQuery?</a></li>
<li><a href="#what-is-in-bigquery"><span class="toc-section-number">1.2</span> What is in BigQuery</a></li>
</ul></li>
<li><a href="#setup"><span class="toc-section-number">2</span> Setup</a></li>
<li><a href="#the-dplyr-package-for-bigquery"><span class="toc-section-number">3</span> The dplyr package for bigquery</a></li>
<li><a href="#example-use-cases"><span class="toc-section-number">4</span> Example use cases</a><ul>
<li><a href="#how-many-variants-of-each-type-are-present-for-each-disease-type"><span class="toc-section-number">4.1</span> How many variants of each type are present for each disease type?</a></li>
</ul></li>
</ul>
</div>

<div id="background" class="section level1">
<h1><span class="header-section-number">1</span> Background</h1>
<div id="what-is-this-bigquery" class="section level2">
<h2><span class="header-section-number">1.1</span> What is this BigQuery?</h2>
<p>BigQuery is Google’s managed SQL-like database system that is designed to run at PB scales. It allows storing, querying, and manipulating absolutely <em>huge</em> datasets with little more effort (perhaps even less) than operating on a standard relational database. Cost is a pay-as-you-go model, based on the amount of data processed.</p>
</div>
<div id="what-is-in-bigquery" class="section level2">
<h2><span class="header-section-number">1.2</span> What is in BigQuery</h2>
<p>The <a href="http://cgc.systemsbiology.net/">Institute for Systems Biology (ISB) Cancer Genomics Cloud</a> have loaded all of the TCGA level 3 data into BigQuery, enabling very facile slicing of data to a very fine granularity. Bulk data in file form is still likely to be the most appropriate way to do large-scale data mining and inference, but for gene- or sample-based queries or to join datasets in novel ways, BigQuery offers a possible solution.</p>
</div>
</div>
<div id="setup" class="section level1">
<h1><span class="header-section-number">2</span> Setup</h1>
<pre class="r"><code>library(bigrquery)</code></pre>
<pre class="r"><code>bigrquery::list_tables(&quot;isb-cgc&quot;, &quot;tcga_201510_alpha&quot;)</code></pre>
<p>The first time you run this, you will be likely asked to login to a google billing account. This process will involve R kicking you over to a web browser (the first time) where you will need to login to your google cloud project. You will likely see this type of message….</p>
<pre><code>Use a local file to cache OAuth access credentials between R sessions?
1: Yes
2: No

Selection: Yes
Adding ^\.httr-oauth$ to .Rbuildignore
Waiting for authentication in browser...
Press Esc/Ctrl + C to abort
Authentication complete.</code></pre>
<p>Assuming that you have set up an account with the ISB Cancer Genomics Cloud Pilot, you can authorize R to access the account, and from then on will be able to operate without browser interaction.</p>
<pre class="r"><code>bigrquery::list_tables(&quot;isb-cgc&quot;, &quot;tcga_201510_alpha&quot;)</code></pre>
<pre><code>## Auto-refreshing stale OAuth token.</code></pre>
<pre><code>##  [1] &quot;Annotations&quot;            &quot;Biospecimen_data&quot;      
##  [3] &quot;Clinical_data&quot;          &quot;Copy_Number_segments&quot;  
##  [5] &quot;DNA_Methylation_betas&quot;  &quot;Protein_RPPA_data&quot;     
##  [7] &quot;Somatic_Mutation_calls&quot; &quot;mRNA_BCGSC_HiSeq_RPKM&quot; 
##  [9] &quot;mRNA_UNC_HiSeq_RSEM&quot;    &quot;miRNA_expression&quot;</code></pre>
</div>
<div id="the-dplyr-package-for-bigquery" class="section level1">
<h1><span class="header-section-number">3</span> The dplyr package for bigquery</h1>
<p>The <a href="https://cran.r-project.org/web/packages/dplyr/vignettes/databases.html">dplyr database tutorial</a> discusses how to use <code>dplyr</code> with Google BigQuery. We are going to use that approach here to simplify the interaction with the database.</p>
<pre class="r"><code>library(dplyr)</code></pre>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
<p>Next, we set up a connection between <code>dplyr</code> and <code>bigquery</code>.</p>
<pre class="r"><code>my_billing = &quot;isb-cgc-01-0006&quot; # replace billing info here with your own
bq = src_bigquery(&quot;isb-cgc&quot;, &quot;tcga_201510_alpha&quot;, billing = my_billing)</code></pre>
<p>The available tables in the <code>tcga_201510_alpha</code> database are given by simply showing the <code>bq</code> object.</p>
<pre class="r"><code>bq</code></pre>
<pre><code>## src:  bigquery [isb-cgc:tcga_201510_alpha]
## tbls: Annotations, Biospecimen_data, Clinical_data, Copy_Number_segments,
##   DNA_Methylation_betas, miRNA_expression, mRNA_BCGSC_HiSeq_RPKM,
##   mRNA_UNC_HiSeq_RSEM, Protein_RPPA_data, Somatic_Mutation_calls</code></pre>
<p>To work with a table from BigQuery using <code>dplyr</code>, we need to set up a <code>dplyr</code> table connection, one for each table that we wish to use. Here, I do that with the <code>Annotations</code> table.</p>
<pre class="r"><code>tbl_annot = tbl(bq,&#39;Annotations&#39;)
# and show the columns
colnames(tbl_annot)</code></pre>
<pre><code>##  [1] &quot;annotationId&quot;             &quot;annotationCategoryId&quot;    
##  [3] &quot;annotationCategoryName&quot;   &quot;annotationClassification&quot;
##  [5] &quot;annotationNoteText&quot;       &quot;Study&quot;                   
##  [7] &quot;itemTypeName&quot;             &quot;itemBarcode&quot;             
##  [9] &quot;AliquotBarcode&quot;           &quot;ParticipantBarcode&quot;      
## [11] &quot;SampleBarcode&quot;            &quot;dateAdded&quot;               
## [13] &quot;dateCreated&quot;              &quot;dateEdited&quot;</code></pre>
<p>Our <code>dplyr</code> table object behaves essentially like a data.frame when using the <code>dplyr</code> style of coding.</p>
<pre class="r"><code>head(tbl_annot)</code></pre>
<pre><code>## Source:   query [?? x 14]
## Database: bigquery [isb-cgc:tcga_201510_alpha]
## 
##    annotationId annotationCategoryId
##           &lt;int&gt;                &lt;int&gt;
## 1         29663                  203
## 2         29662                   25
## 3         29661                   25
## 4         29660                   29
## 5         29658                   29
## 6         29659                   29
## ..          ...                  ...
## Variables not shown: annotationCategoryName &lt;chr&gt;,
##   annotationClassification &lt;chr&gt;, annotationNoteText &lt;chr&gt;, Study &lt;chr&gt;,
##   itemTypeName &lt;chr&gt;, itemBarcode &lt;chr&gt;, AliquotBarcode &lt;chr&gt;,
##   ParticipantBarcode &lt;chr&gt;, SampleBarcode &lt;chr&gt;, dateAdded &lt;chr&gt;,
##   dateCreated &lt;chr&gt;, dateEdited &lt;chr&gt;.</code></pre>
</div>
<div id="example-use-cases" class="section level1">
<h1><span class="header-section-number">4</span> Example use cases</h1>
<pre class="r"><code>tbl_mut = tbl(bq,&#39;Somatic_Mutation_calls&#39;)</code></pre>
<div id="how-many-variants-of-each-type-are-present-for-each-disease-type" class="section level2">
<h2><span class="header-section-number">4.1</span> How many variants of each type are present for each disease type?</h2>
<pre class="r"><code>mut_counts = tbl_mut %&gt;% 
  select(Study, Variant_Classification) %&gt;% 
  group_by(Study,Variant_Classification) %&gt;% 
  summarize(n = n()) %&gt;% 
  arrange(Study,Variant_Classification)</code></pre>
<p>```{r</p>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>## R version 3.3.1 (2016-06-21)
## Platform: x86_64-apple-darwin13.4.0 (64-bit)
## Running under: OS X 10.11.6 (El Capitan)
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] dplyr_0.5.0       bigrquery_0.3.0   BiocStyle_2.0.2   yaml_2.1.13      
## [5] rmarkdown_0.9.6.9
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.5     digest_0.6.9    assertthat_0.1  R6_2.1.2       
##  [5] jsonlite_1.0    DBI_0.4-1       formatR_1.4     magrittr_1.5   
##  [9] evaluate_0.9    httr_1.2.1      stringi_1.1.1   lazyeval_0.2.0 
## [13] curl_0.9.7      tools_3.3.1     stringr_1.0.0   rsconnect_0.4.3
## [17] htmltools_0.3.5 openssl_0.9.4   knitr_1.13      tibble_1.0</code></pre>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
