---
title: "SBCGC-R-CreateWorkflow"
author: "Durga Addepalli"
date: "May 2, 2016"
output: html_document
vignette: >
  %\VignetteIndexEntry{SBCGC R Create Workflow}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


```{r include=FALSE}
knitr::opts_chunk$set(eval = FALSE)
library(sevenbridges)
```

# INTRODUCTION

This guide will get you started to use Seven Bridges API with the R client package sevenbridges, and guide you through the steps needed to run the "RNA-seq Alignment - STAR" and RNA-Seq alignment pipeline on the Seven Bridges Genomics platform.

The following primary steps will be included:

* uploading files and setting the metadata;
* running the Picard SAMtoFASTQ commandline tool;
* running the RNA-seq Alignment on STAR; 
* downloading the results.

# First Steps

First download the [R markdown file]
(https://raw.githubusercontent.com/tengfei/sevenbridges/master/vignettes/bioc-workflow.Rmd) of this page, so you can load it to yoru Rstudio 
or your favorite R IDE to run through all examples and tweak setup.

```{r}
download.file("https://github.com/teamcgc/teamcgc.github.io/....SBCGC-R-CreateWorkflow.Rmd", destfile = "~/bioc-workflow.Rmd")
```

## Installing the Package

To download and install the sevenbridges package from Bioconductor, type the following commands in R:

```{r}
source("http://bioconductor.org/biocLite.R")
biocLite("sbgr")
```

Alternatively, you can install the cutting-edge development version of the sbgr package from GitHub:

```{r}
if(!require("devtools", quietly = TRUE)){
    install.packages("devtools") 
}
## Install from github for development version  
source("http://bioconductor.org/biocLite.R")
devtools::install_github("tengfei/sevenbridges", build_vignettes=TRUE, 
  repos=BiocInstaller::biocinstallRepos(),
  dependencies=TRUE)
```

After the installation you can always browser vignette

```{r}
browseVignettes(package = 'sevenbridges')
```

### Register on NCI Cancer Genomics Cloud

![cgc-home](https://teamcgc.nci.nih.gov.s3.amazonaws.com/Images/CGC-Home.png)

You can find login/registration on NCI [Cancer Genomics Cloud homepage](http://www.cancergenomicscloud.org/), follow the [signup tutorial](http://docs.cancergenomicscloud.org/docs/sign-up-for-the-cgc) if you have an ERA Commons or NIH account.

### Get your authentifiation

After you login, you can get your authentication under your account setting and 'developer' tab [tutorial](http://docs.cancergenomicscloud.org/docs/get-your-authentication-token)

## Quickstart

The final goal is make a workflow that accepts one file per sample (or two files for paired-end data), GTF file,genome Fasta files and generate aligned reads, de novo canonical junctions, non-canonical splices, and chimeric (fusion) transcripts. 

The final workflow looks like this, it's composed of two tools: Picard SAM to Fastq command line tool and STAR alignment tool.

![quickstart-flow](https://teamcgc.nci.nih.gov.s3.amazonaws.com/Images/RNA-Seq_AlignmentSTAR.png)

### Create a project on your account via API R client

Now lets use the 'sevenbridges' packages you just installed to create a project.

First step to do is to create an _Auth_ object, almost everything starts from this object. SB-CGC API client follows a pattern like "Auth$properties$action".

On the SB platform/CGC GUI, Auth is your account, and it contains projects, billing groups, users, project contains tasks, apps, files etc, so it's easy to imagine your API call.

To create Auth, just pass token and url, by default url is set to CGC.  
To create an Auth object, run the command below and replace "fake_token" with your own token.

```{r}
a <- Auth(token = "fake_token", url = "https://cgc-api.sbgenomics.com/v2/")
```

Alternatively you can save your authentication cross different platforms in a configuration file called ".sbg.auth.yml" at home folder. This allow you to manage multiple users on multiple platforms.

To create a project, you need to know your billing group id, cost related to this project will be charged from this billing group.

```{r}
(b <- a$billing())
## a single billing group is shown
```

Now let's create a new project called "IRP-CGC-with-R", save it to a 'p' object for convenient usage for any call related to this project.

```{r}
(p <- a$project_new("IRP-CGC-with-R", 
                    billing_group_id = b$id, 
                    description = "This project is for IRP-CGC R-Workshop"))
```

Now check it on CGC, you will see a fresh new project is created. 

![NewProject](https://teamcgc.nci.nih.gov.s3.amazonaws.com/Images/NewProject.png)

### List all Projects
This call returns a list of all projects you are a member of. Each project’s project_id and URL on the CGC will be returned.
```{r}
a$project()
```
Then if you want to list the projects owned by and accessible to a particular user, specify the owner argument. Each project’s ID and URL will be returned.

```{r}
a$project(owner = "Durga")
```

To get details about project(s), use detail = TRUE

```{r}
a$project(detail = TRUE)
```

### Partial Match Project Name
SB CGC also supports partial name match in this interface. The first argument for the call is “name”, users can provide part of the name.
```{r}
a$project("hello")
```

### Delete a Project
Only single project could be deleted by call $delete(), so please pay attention to the returned object from a$project(), sometimes if you are using partial matching by name, it will return a list. 

To delete it, just call, but I will suggest you keep it for following tutorial : ) 

```{r}
## remove it, not run
a$project("IRP-CGC-with-R")$delete()
## check
## will delete all projects matcht the name
delete(a$project("IRP-CGC-with-R_donnot_delete_me"))
```


### List all Files
This call returns a list of all files in a specified project that you can access. For each file, the call returns: 
* Its ID,
* Its filename. 
The project is specified as a query parameter in the call.
To list all the files in the current project:
```{r}
p$file()
```

To list all files belonging to a project (Workshop Demo):
```{r}
n <- a$project(name = "Workshop Demo")
a$file(project = n$id)
a$file("Readspergene", project = n$id, detail = TRUE)
```


### Copy a file or group of files

This call copies the specified file to a new project. Files retain their metadata when copied, but may be assigned new names in their target project.

Note that Controlled Data files may not be copied to Open Data projects. To make this call, you should have copy permission within the project you are copying from.

Let’s try to copy a file from CGC public files, the id you can tell from the url is “561e1b33e4b0aa6ec48167d7”

You must provide

* id file id, or list/vector of files ids.
* project parameter: project id.
* name is optional, if omitted, use the same.

To copy a single file
```{r}
fid <- "56d5dd4be4b038b4e319be9b"
a$project("Workshop Demo")$file(id = fid)$copyTo(pid)
```
To Copy multiple files
```{r}
fid <- "56d5dd4be4b0f636ff14cb69"
fid2 <- "56d5dd4be4b0ac987e61af8e"
pid <- a$project("Workshop Demo")$id
a$copyFile(c(fid, fid2), project = pid)
p$file()
```

### Delete file(s)

Note: the delete action only works for single file now, make sure your file call returns a single file not a file list.
```{r}
## Delete file
a$project("IRP-CGC-with-R")$file()[[1]]$delete()
## confirm the deletion
a$project("IRP-CGC-with-R")$file()

## Search for specific file by pattern match and delete, this can delete multiple files too.
a$project("IRP-CGC-with-R")$file("Readspergene")$delete()

```

## Lets Copy some Apps into our Project IRP-CGC-with-R
### First lets search for the Apps
You can list all Apps available to you.
```{r}
a$app()
## or show details
a$app(detail = TRUE)
```

You can serach by name, partial name or ID
```{r}
## pattern match
a$app(name = "get_http")

## unique id
a$app(id = "tsang/temp/txfr")

## get a specific revision from an app
a$app(id = aid, revision = 0)
```

To list all apps belong to one project use project argument
```{r}
a$project("Workshop Demo")$app()

## or alternatviely
pid <- a$project("Workshop Demo")$id
a$app(project = pid)
```

To run this workflow, we need two apps in our project:
* get-http_file
* RNA-Seq Alignment STAR

### Copy these Apps to your Project
This call copies the specified app to the specified project. The app should be one in a project that you can access; this could be an app that has been uploaded to the CGC by a project member, or a publicly available app that has been copied to the project.

Need two arguments
* project: id character
* name: optional, to re-name your app

Copy app "get-http_file" 
```{r}
app.gets3fl <- a$copyApp(id =  "tsang/temp/txfr/2", project = pid, name = "copytxfrv2")
apid <- app.gets3fl$id
## check it is copied
a$app(project = pid)
```


Search and copy the PUBLIC workflow RNA-Seq Alignment STAR
```{r}
a$app("STAR", visibility = "public", complete = TRUE)

app.rnawf <- a$copyApp(id =  "djordje_klisic/public-apps-by-seven-bridges/rna-seq-alignment-star/3", project = pid, name = "star-wfl-copy")

rwid <- app.rnawf$id
## check it is copied
a$app(project = pid)
```


### Upload files from Amazon bucket to our project

Here we demonstrate how to upload files directly from a S3 bucket to your CGC project. We will be using the Get_Http_file app to get files. ......

To create a draft task, you need to call the task_add function from Project object. And you need to pass following arguments

name: name for this task
description: description for this task
app: app id you have access to
inputs: inputs list for this task

```{r}
p$task_add(name = "cptxfrv2testrun", 
           description = "Using get_http_file app to get files from S3 bucket", 
           app=apid,
           inputs = list(url = "http://teamcgc.nci.nih.gov.s3-website-us-east-1.amazonaws.com/20160325SBGworkshop/FastqFiles/G28029_pe_1.fastq", ofname = "G28029_pe_1.fastq"))

## confirm
p$task(status = "draft")
```

### Run the task

This call runs (executes) the specified task. Only tasks whose status is “DRAFT” may be run.

```{r}
tsk <- p$task(status = "draft")
tsk$run()
## run update without information just return latest information
tsk$update()
```


### Run the RNA-Seq Alignment STAR Workflow

Now we will run the RNA-seq Alignment STAR workflow (on which we worked in the last workshop) taking the BAM file uploaded from the previous task, as an input.

WorkFlow Description:

1. What tools are inbuilt in this workflow?
    - Picard Sam to FastQ (Tool)
    - SBG FASTQ Quality Detector: FASTQ Quality Scale Detector detects which quality encoding scheme       was used in your reads and automatically enters the proper value in the "Quality Scale"             metadata field.
    - STAR Genome Generate: STAR Genome Generate is a tool that generates genome index files. One         set of files should be generated per each genome/annotation combination. Once produced, these       files could be used as long as genome/annotation combination stays the same. Also, STAR Genome       Generate which produced these files and STAR aligner using them must be the same toolkit            version.
    - Picard SortSam: Picard SortSam sorts the input SAM or BAM. Input and output formats are             determined by the file extension.
    - RNA-seq Alignment - STAR (Workflow)
  
2. What Input, Output and Parameter you want to run the workflow
   - Input 
      - GTF file
      - BAM File
      - Genome Fasta Files

    - Output
      - Aligned reads (BAM)
      - Unmapped reads (fastQ)
      - Splice Junctions
      - Reads per Gene (TAB)
      - Chimeric Junctions (Junction)
      - Chimeric Alignments (SAM)
      - Aligned Sorted by Coord (BAI)
      - Log Files
      - Unpaired (fastQ)
      - First Strand (fastQ)
      - Second Strand (fastQ)
      

### Running the RNA-Seq Alignment STAR which takes a FASTQ file as input

```{r}
fastq1 <- p$file(name = "G28029_pe_1.fastq")
fastq2 <- p$file(name = "G28029_pe_2.fastq")


## Set metadata / Pairend info

fastq1$setMeta(Paired_end = "1")
fastq2$setMeta(Paired_end = "2")


(fastq.in <- p$file(name = "G28029"))
(gtf.in <- p$file(".gtf"))
(gtf.in1 = new('FilesList',listData=list(gtf.in)))
(fasta.in <- p$file("HG19_Broad_variant.fasta"))

## Check the input parameters' Class
## inputs : list(`sbg:y` = 195.083310633897, id = "#sjdbGTFfile", type = list("null", list(items = "File", type = "array")), `sbg:x` = 160.499975860118, label = "sjdbGTFfile")list(`sbg:y` = 323.74995018542, id = "#fastq", `sbg:x` = 164.24999140203, label = "fastq", type = list(list(items = "File", type = "array")), `sbg:includeInPorts` = TRUE)list(`sbg:y` = 469.999910578135, id = "#genomeFastaFiles", type = list("File"), `sbg:x` = 167.749960079791, label = "genomeFastaFiles")

class(fastq.in)
class(gtf.in)
class(fasta.in)

tsk = p$task_add(name = "rwfltest", 
           description = "Testing the RNA-seq STAR workflow", 
           app = rwid,
           inputs =  list(fastq = fastq.in, 
                          genomeFastaFiles = fasta.in,
                          sjdbGTFfile = list(gtf.in1)))


tsk2 = p$task_add(name = "rwfltest_SB:FL", 
             description = "Testing the RNA-seq STAR workflow", 
            app = rwid,
            inputs =  list(fastq = fastq.in, 
                          genomeFastaFiles = fasta.in,
                         sjdbGTFfile = sevenbridges:::FilesList(gtf.in)))

## confirm
p$task(status = "draft")

## Run the Task
tsk1 <- p$task(status = "draft")
tsk1$run()
tsk1$update()
tsk1$monitor()
```

### Running the RNA-Seq Alignment STAR which takes a BAM file as input

```{r}
## Copy the app
app.rnawfbam <- a$copyApp(id = "Durga/test1-tumor-normal/rna-seq-alignment-star-baseline-20160217", project = pid, name = "star-wfl-inbam")
rwbid <- app.rnawfbam$id
## check it is copied
a$app(project = pid)

## Run the app

(bam.in <- p$file("G41707Downsample_data_36_downsampled.bam"))
(gtf.in <- p$file(".gtf"))
(fasta.in <- p$file("HG19_Broad_variant.fasta"))

tsk3 = p$task_add(name = "rwfl-inbamtest2-R", 
           description = "Testing the RNA-seq STAR workflow with input BAM file", 
           app = rwbid,
           inputs =  list(input_file = bam.in, 
                          genomeFastaFiles = fasta.in,
                          sjdbGTFfile = sevenbridges:::FilesList(gtf.in)))

## confirm
p$task(status = "draft")

## Run the Task
tsk3 <- p$task(status = "draft")
tsk3$run()
tsk3$update()
tsk3$monitor()
```

## BATCH - Running the RNA-Seq Alignment STAR which takes multiple BAM file as input
```{r}
## Copy the app
## another app for batch
app.rnawfbamb <- a$copyApp(id = "Durga/rworkshop2/rna-seq-alignment-star-baseline-20160217", project = pid, name = "star-wf-inbam-batch-UIcopy")

rwbbid <- app.rnawfbamb$id
## check it is copied
a$app(project = pid)


## Run the app

(bam.in <- p$file("downsampled.bam"))
(gtf.in <- p$file(".gtf"))
(fasta.in <- p$file("HG19_Broad_variant.fasta"))

tsk3 = p$task_add(name = "rwf-inbam_batch-R", 
           description = "Testing the RNA-seq STAR workflow with multiple input BAM files", 
           app = rwbbid,
           batch = batch(input = "input_file"),
           inputs =  list(input_file = bam.in, 
                          genomeFastaFiles = fasta.in,
                          sjdbGTFfile = sevenbridges:::FilesList(gtf.in)))



## confirm
p$task(status = "draft")

## Run the Task
tsk3 <- p$task(status = "draft")
tsk3$run()
tsk3$update()
tsk3$monitor()
```


## More tutorials

After you install the package

```{r}
browseVignettes("sevenbridges")
```

or on Bioconductor devel branch sevenbridges [page](http://bioconductor.org/packages/3.3/bioc/html/sevenbridges.html)

